api/lib/opentelemetry/trace/samplers/always_sample_sampler.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/trace/samplers/basic_decision'
+
+module OpenTelemetry
+  module Trace
+    module Samplers
+      # The {AlwaysSampleSampler} always returns a true sampling decision.
+      class AlwaysSampleSampler
+        ALWAYS_SAMPLE_DECISION = BasicDecision.new(decision: true)
+
+        # Returns the sampling {Decision} for a {Span} to be created
+        #
+        # @param [SpanContext] span_context The {SpanContext} of a parent span,
+        #   typically extracted from the wire. Can be nil.
+        # @param [Boolean] extracted_context True if span_context was extracted
+        #   from the wire. Can be nil.
```

Comment:
It is not clear from the specification that this can be `nil`.

Reply:
The [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-sampling.md#shouldsample) says:

> SpanContext of a parent Span. Typically extracted from the wire. Can be null.

Reply:
Oh, you're asking about the `extracted_context`. I suppose that is ambiguous. I cross referenced the Java Decision Interface, where it is also `nullable`: 

https://github.com/open-telemetry/opentelemetry-java/blob/6d1b31bfa639af653649d694ea5660fb7c3fd5ce/api/src/main/java/io/opentelemetry/trace/util/Samplers.java#L100-L108

Reply:
👍 would be good to add the comment from there: `nil if this is a root span`.

Reply:
I went ahead and qualified the nilness of `span_context` and `extracted_context`. `extracted_context` might get added to span context: https://github.com/open-telemetry/opentelemetry-specification/pull/216, which would clean this up further.